packages/router/router.ts

```diff
@@ -1971,6 +1973,7 @@ export function createRouter(init: RouterInit): Router {
 
     // If the loader threw a redirect Response, start a new REPLACE navigation
     if (isRedirectResult(result)) {
+      fetchRedirectIds.add(key);
```

Comment:
Track `fetcher.load` redirects as well so they can get cleaned up on completion of the redirect navigation